src/coreclr/tools/aot/Mono.Linker.Tests.Cases/RequiresCapability/RequiresViaDataflow.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.RequiresCapability
+{
+	[SkipKeptItemsValidation]
+	[ExpectedNoWarnings]
+	class RequiresViaDataflow
+	{
+		// Base/Derived and Implementation/Interface differs between linker and analyzer https://github.com/dotnet/linker/issues/2533
+		[ExpectedWarning ("IL2026", "--DynamicallyAccessedTypeWithRequires.MethodWithRequires--")]
+		[ExpectedWarning ("IL2026", "TypeWhichOverridesMethod.VirtualMethodRequires()", "--TypeWhichOverridesMethod.VirtualMethodRequires--", ProducedBy = ProducedBy.Analyzer | ProducedBy.NativeAot)]
```

Comment:
Do we have a bug tracking the NativeAOT side of this? Or this really a linker-only bug (probably due to override resolution being broken in Cecil) - probably worth a comment either way.

Reply:
The issue is marked in the comment. It doesn't have to do with Cecil, by choice we decide to print the warnings in a particular way. Basically for linker no matter if we use the base or derived/interface or implementation, we always print a warning in the base/interface because IL only points to base/interface. Analyzer implementation got more complicated because it does point to the derived/implementation and interface resolution based on implementation was expensive. NativeAOT just happens to use the same approach as the analyzer.